### PR TITLE
(prod) Fix for account id missing in msal get account

### PR DESCRIPTION
**Pull request by shahzaibj**
Related: https://github.com/AzureAD/ad-accounts-for-android/pull/1785

Related: https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1560

### CODE REVIEW

**Review by AdamBJohnsonx — APPROVED** [review]
This is really creating a null-value-object to avoid the crash.

File: msal/src/main/java/com/microsoft/identity/client/Account.java [hunk 1]
```diff
@@ -93,6 +96,23 @@ public String getId() {
...
+        if (StringUtil.isEmpty(id)) {
+            com.microsoft.identity.common.logging.Logger.warn(
+                    TAG + methodName,
+                    "Unable to get account id from either ClientInfo or IdToken. Attempting to obtain from home account id."
+            );
+            id = AccountUtil.getUIdFromHomeAccountId(mHomeAccountId);
+        }
+
+        if (StringUtil.isEmpty(id)) {
+            com.microsoft.identity.common.logging.Logger.warn(
```
Comment (AdamBJohnsonx) [inline msal/src/main/java/com/microsoft/identity/client/Account.java:108]: Should this be an error?

Comment (shahzaibj) [reply msal/src/main/java/com/microsoft/identity/client/Account.java:108]: probably..

Comment (shahzaibj) [reply msal/src/main/java/com/microsoft/identity/client/Account.java:108]: I can make that change

Comment (shahzaibj) [reply msal/src/main/java/com/microsoft/identity/client/Account.java]: Addressed https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1558/commits/c6748c3d7df139933f1629866314a277dca2b1dd

File: msal/src/main/java/com/microsoft/identity/client/Account.java [hunk 2]
```diff
@@ -93,6 +96,24 @@ public String getId() {
...
+                    "Unable to get account id from either ClientInfo or IdToken. Attempting to obtain from home account id."
+            );
+            id = AccountUtil.getUIdFromHomeAccountId(mHomeAccountId);
+        }
+
+        if (StringUtil.isEmpty(id)) {
+            com.microsoft.identity.common.logging.Logger.error(
+                    TAG + methodName,
+                    "Account ID is empty. Returning MISSING_FROM_THE_TOKEN_RESPONSE.",
+                    new NullPointerException("Account id is null or empty")
```
Comment (AdamBJohnsonx) [inline msal/src/main/java/com/microsoft/identity/client/Account.java:111]: you could pass null for this exception, IIRC... ;-)

Comment (shahzaibj) [reply msal/src/main/java/com/microsoft/identity/client/Account.java]: Done: https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1558/commits/6a560eacd0d8ba33606dec559aa3f03ffe8958fb

**Pull request merged**